src/CBLDatabase_Internal.hh

```diff
@@ -356,7 +357,16 @@ private:
 
     Retained<CBLDocument> _getDocument(slice docID, bool isMutable, bool allRevisions) const {
         C4DocContentLevel content = (allRevisions ? kDocGetAll : kDocGetCurrentRev);
-        Retained<C4Document> c4doc = _c4db.useLocked()->getDocument(docID, true, content);
+        Retained<C4Document> c4doc = nullptr;
+        try {
+            c4doc = _c4db.useLocked()->getDocument(docID, true, content);
+        } catch (litecore::error& e) {
+            if (e == litecore::error::BadDocID) {
+                CBL_Log(kCBLLogDomainDatabase, kCBLLogWarning,
+                        "Invalid document ID '%.*s' used", FMTSLICE(docID));
+                return nullptr;
+            }
```

Comment:
For other errors that are not baddocid this should still throw.

Reply:
Right, forgot that.